sos/report/plugins/networking.py

```diff
@@ -230,38 +229,13 @@ def setup(self):
 
         # Capture additional data from namespaces; each command is run
         # per-namespace.
-        ip_netns = self.collect_cmd_output("ip netns")
+        self.add_cmd_output("ip netns")
         cmd_prefix = "ip netns exec "
-        if ip_netns['status'] == 0:
-            out_ns = []
-            # Regex initialization outside of for loop
-            if self.get_option("namespace_pattern"):
-                pattern = '(?:%s$)' % '$|'.join(
-                        self.get_option("namespace_pattern").split()
-                        ).replace('*', '.*')
-            for line in ip_netns['output'].splitlines():
-                # If there's no namespaces, no need to continue
-                if line.startswith("Object \"netns\" is unknown") \
-                        or line.isspace() \
-                        or line[:1].isspace():
-                    continue
-                # if namespace_pattern defined, append only namespaces
-                # matching with pattern
-                if self.get_option("namespace_pattern"):
-                    if bool(match(pattern, line)):
-                        out_ns.append(line.partition(' ')[0])
-
-                # if namespaces is defined and namespace_pattern is not defined
-                # remove from out_ns namespaces with higher index than defined
-                elif self.get_option("namespaces") != 0:
-                    out_ns.append(line.partition(' ')[0])
-                    if len(out_ns) == self.get_option("namespaces"):
-                        self._log_warn("Limiting namespace iteration " +
-                                       "to first %s namespaces found"
-                                       % self.get_option("namespaces"))
-                        break
-                else:
-                    out_ns.append(line.partition(' ')[0])
+        if self.commons['namespaces']['network']:
+            out_ns = self.filter_namespaces(
+                        self.commons['namespaces']['network'],
+                        self.get_option("namespace_pattern"),
+                        self.get_option("namespaces"))
 
             for namespace in out_ns:
```

Comment:
This would all be condensed down to the following for example with the changes mentioned above

```python
for namespace in self.get_namespaces(self.get_option("namespace_pattern"),
                                     self.get_option("namespaces")):
    [... collections ...]
```

Reply:
I was lazy and kept `if` and `out_ns` to avoid the large indentation changes, and due to the other use of `out_ns` for `ethtool_namespaces` option.
I'll update both more properly this time. :)